### PR TITLE
Upload:: Honor OUTPUT_EXT

### PR DESCRIPTION
**Pull request by ccli8**
### Summary of changes 

This PR changes upload method `gen_upload_target` signature from:
```
function(gen_upload_target TARGET_NAME BIN_FILE HEX_FILE)
```
To:
```
function(gen_upload_target TARGET_NAME BINARY_FILE)
```

If `OUTPUT_EXT` is undefined or defined to "bin", BINARY_FILE is .bin file, otherwise .hex file.

The update applies to all UploadMethodXxx files.

Replace #329 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by multiplemonomials** [comment]
This is a good start but we will also need to update all of the UploadMethodXXXX files under mbed-os/tools/cmake/upload_methods to use this new signature.

File: tools/cmake/upload_methods/UploadMethodOPENOCD.cmake [hunk 1]
```diff
@@ -39,7 +39,7 @@ if { [adapter name] == \"hla\" } {
 	set(OPENOCD_ADAPTER_SERIAL_COMMAND -f ${CMAKE_BINARY_DIR}/openocd_adapter_config.cfg CACHE INTERNAL "" FORCE)
 endif()
 
-function(gen_upload_target TARGET_NAME BIN_FILE)
+function(gen_upload_target TARGET_NAME BINARY_FILE)
 
 	# unlike other upload methods, OpenOCD uses the elf file
 	add_custom_target(flash-${TARGET_NAME}
```
Comment (multiplemonomials) [inline tools/cmake/upload_methods/UploadMethodOPENOCD.cmake:45]: We should update line 51 here to just use `${BINARY_FILE}` and simplify that logic.

**Comment by multiplemonomials** [comment]
Looks good just one fix ^

File: tools/cmake/upload_methods/UploadMethodOPENOCD.cmake (hunk 1 above)
Comment (ccli8) [reply tools/cmake/upload_methods/UploadMethodOPENOCD.cmake:45]: yes, consistent with other upload methods, just see passed-in `BINARY_FILE`

**Review by multiplemonomials — APPROVED** [review]

**Pull request merged**